core/types.h

```diff
@@ -139,6 +139,14 @@ enum HollyInterruptID
 int darw_printf(const char* Text,...);
 #endif
 
+#if defined(__APPLE__) && defined(__MACH__) && HOST_CPU == CPU_ARM64
+	#define __ARM_MAC__
+	#include "pthread.h"
+	static void WriteProtect(bool enabled) { pthread_jit_write_protect_np(enabled); }
```

Comment:
I would prefer a more descriptive name, perhaps simply `jitWriteProtect()`

Also `inline static` is better to avoid compilation warnings (function defined but not used...)